Add Starlight 0.28 blog post (#1257)
withastro/astro.build: src/content/blog/starlight-028.mdx

coverImage: '/src/content/blog/_images/starlight-028/blog-post-starlight-v0.28.webp'
socialImage: '/src/content/blog/_images/starlight-028/og-starlight-v0.28.webp'
---

**Starlight v0.28 is out now! The latest version of Astro's official, all-in-one documentation site builder just dropped. Let’s take a look at some of the highlights from recent Starlight releases.**

- [Next-level localization](#next-level-localization)
- [On-demand rendering support](#on-demand-rendering-support)
- [Tabs that remember you](#tabs-that-remember-you)
- [Sidebar state persistence](#sidebar-state-persistence)
- [Simpler sidebar configuration](#simpler-sidebar-configuration)
- [Sparkling fresh docs](#sparkling-fresh-docs)
- [A Markdoc preset for the adventurous](#a-markdoc-preset-for-the-adventurous)

To upgrade an existing project, run the automated `@astrojs/upgrade` CLI from your terminal:

```sh
npx @astrojs/upgrade
```

## Next-level localization

Starlight has been multilingual from day one and v0.28 takes this to the next level.
Starlight’s built-in localization system is now powered by the first-in-class [`i18next`](https://www.i18next.com/) library and available to use anywhere in your site.

> Use the new `Astro.locals.t()` method from any Astro component to render Starlight’s built-in UI labels or your own custom strings, including support for interpolation and pluralization:

Author: Ah yeah, the new docs weren’t out yet. Will add a link now!

This is a progressive enhancement — if JavaScript isn’t available on a user’s device, the sidebar renders as usual ensuring a robust experience for all.

## Simpler sidebar configuration

In earlier versions of Starlight, setting up your [sidebar config](https://starlight.astro.build/guides/sidebar/) could get a little verbose:

```js title="astro.config.mjs" {4-6}
starlight({
	title: 'Old Skool Starlight',
	sidebar: [
		{ label: 'Page one', link: '/one' },
		{ label: 'Page two', link: '/two' },
		// ...
	],
}),
```

With Starlight v0.25, this got a lot simpler for internal links.
You can now specify just the slug of the page you’re linking to and Starlight will use the page title as the link label automatically:

```js title="astro.config.mjs" {4}
starlight({
	title: 'Cutting Edge Starlight',
	sidebar: [
		'one', 'two', // ...
	],
}),
```

As a bonus, with this syntax you get link checking for free — no more accidental 404s! Thanks to [Lorenzo](https://github.com/lorenzolewis) for adding this feature.

## Sparkling fresh docs

> And we've been working on our own docs, too!
> We overhauled the documentation for Starlight’s built-in components.
> Each component now has a dedicated page including common usage examples and a full prop reference.

Reviewer: ```suggestion
We've been working hard on our own docs, too!
Overhauling all the documentation for Starlight’s built-in components.
Now, each component has their own dedicated page including common usage examples and a fully documented `prop` references.
```

Visit the new [“Using components” guide](https://starlight.astro.build/components/using-components/) to explore the available components.

Of course, these wouldn’t be Astro docs without translations.

> Shout out to `@dreyfus92`, `@jsparkdev`, `@dragomano`, `@liruifengv`, `@zhoushengdao`, `@Nin3lee`, `@casungo`, `@at-the-vr`, and `@thomasbnt` for keeping our Spanish, Korean, Russian, Simplified Chinese, Italian, Hindi, and French docs updated!

Reviewer: can we link to github profiles?

## A Markdoc preset for the adventurous

Starlight users interested in Markdoc can use the new `@astrojs/starlight-markdoc` package to quickly configure all of Starlight’s built-in components for use in Markdoc files.
See our [Markdoc set-up guide](https://starlight.astro.build/guides/authoring-content/#markdoc) for more details.

## Try Starlight today!

> We’ve been excited to see companies like [Cloudflare](https://developers.cloudflare.com/), [Crowdin](https://support.crowdin.com/), [StackBlitz](https://tutorialkit.dev/), [Sentry](https://spotlightjs.com/), [Font Awesome](https://docs.fontawesome.com/), and [SST](https://sst.dev/docs/) embrace Starlight when building docs. Join them and open-source projects like [FreeCodeCamp](https://contribute.freecodecamp.org/), [RealWorld Apps](https://www.realworld.how/), [Biome](https://biomejs.dev/), [Jimp](https://jimp-dev.github.io/jimp/), [Shepherd](https://docs.shepherdpro.com/), [Bloc](https://bloclibrary.dev/), and [Knip](https://knip.dev/) — to name just a few! — and get started today.

Reviewer: 💚 I loved reading this, wow 🤯

Visit the [Starlight docs](https://starlight.astro.build/) or create a new site right from your terminal:

```sh
npm create astro@latest -- --template starlight
```

## Previously…

import RelatedPosts from './_components/RelatedPosts.astro';

> <RelatedPosts slugs={['starlight-turns-one', 'wp-engine']} />

Reviewer: ques que say?
